### ISSUE TITLE
Glitch effect bug

### ISSUE THREAD

**Issue by Ryosama2** [n00mkrad/flowframes#183]
My log here:
[glitch effect report.txt](https://github.com/n00mkrad/flowframes/files/9761789/glitch.effect.report.txt)

Dear developers,

When I review the output of Flowframes software, my output video had a glitch effect issue.
And this is the video I had done for frame rate boosting: https://youtu.be/CJ0djllyqwY

I have tried several ways to deal with this issue:
**_- Reset all settings
- Clear cache (clear model cache)
- Restart my pc_**

**But none of them works**

![2022-10-12_073118](https://user-images.githubusercontent.com/115549537/195258274-8f026d20-58f7-45b1-88da-05c367472050.jpg)

![2022-10-12_073200](https://user-images.githubusercontent.com/115549537/195258263-f8c19f7c-74ba-4298-913c-2c314bae01ce.jpg)
![2022-10-12_073221](https://user-images.githubusercontent.com/115549537/195258266-2a1c3dae-dc12-4df2-b1cf-42e851645c13.jpg)
![2022-10-12_122235](https://user-images.githubusercontent.com/115549537/195258269-e602b544-9fd8-4764-96b1-dd0d0e0c198f.jpg)
![2022-10-12_122305](https://user-images.githubusercontent.com/115549537/195258271-6ad6aac8-2df7-47d7-88b5-30f54d0a2522.jpg)

**Comment by Ryosama2** [comment]
![2022-10-12_073118-1](https://user-images.githubusercontent.com/115549537/195258994-683a3e40-b4cb-451a-bb4a-1bb04e3395f9.jpg)
![2022-10-12_073118](https://user-images.githubusercontent.com/115549537/195258999-feb634bd-65d9-4ac8-9774-4c39e69d0ee3.jpg)
**Glitch effect bug**

**Comment by Ryosama2** [comment]
![2022-10-12_143825](https://user-images.githubusercontent.com/115549537/195281035-b6d71eaa-d491-4731-b2d9-a2f3150c3d88.jpg)

Glitching again

**Comment by n00mkrad** [comment]
Does this happen to a specific video or to all

**Comment by Ryosama2** [comment]
> Does this happen to a specific video or to all

Currently, it happens to all video that I am going to increase frame rate by Flkowframes.

Here are some video that I am currently using to increase frame rate with Flowframes, and the output had glitch effects.

- https://youtu.be/CJ0djllyqwY
- https://youtu.be/8m1l3CWETJw
- https://youtu.be/k0n8Yl7ZdKo
- https://youtu.be/AQnpQSNOXDs

They are 4 videos that Flowframes had the glitching problem when the output has finished.

If you want to look further to see my error glitching video, here is the link that it has glitching effect when I had done the flowframes: https://thptthaininhh-my.sharepoint.com/:v:/g/personal/minhhlt2000_1driver_vip/EZavJEdCoTpFpn_HIK91xOoBjcmsITYI_0GHLcU8anYbGw?e=kqCFe3
![2022-10-12_173226](https://user-images.githubusercontent.com/115549537/195321113-7a1a752c-f7f6-4b2f-b678-84a2359c8d8e.jpg)
![2022-10-12_122235](https://user-images.githubusercontent.com/115549537/195321122-792481c2-84c5-4ef7-b1fa-c8135f28f24d.jpg)
![2022-10-12_122305](https://user-images.githubusercontent.com/115549537/195321125-b47293be-f515-4651-9c61-36de628516f8.jpg)
![2022-10-12_143825](https://user-images.githubusercontent.com/115549537/195321130-c30ae70a-c273-4856-9424-a0c2ca4355a3.jpg)
![2022-10-12_172632](https://user-images.githubusercontent.com/115549537/195321133-148f2a78-1273-457d-be05-292d65d8035e.jpg)

I mean, if you look at the original video in Youtube here: https://youtu.be/CJ0djllyqwY ; there is no glitching at all. But the Flowframes software has this problem.

Thanks for taking your time to debug this software. ❤️

**Comment by Ryosama2** [comment]
Glitch effects

![2022-10-12_173755](https://user-images.githubusercontent.com/115549537/195322625-77c1d622-1f52-4c99-adc1-95a7d2219b65.jpg)
![2022-10-12_173856](https://user-images.githubusercontent.com/115549537/195322633-364a9db2-5f52-441e-ab5d-1f81738936bd.jpg)
![2022-10-12_173930](https://user-images.githubusercontent.com/115549537/195322636-e27b0c7b-1e15-48cf-baad-bf0eefe0949d.jpg)

**Comment by Ryosama2** [comment]
Dear developers,

Sorry for bothering you guys and I did not mean to ask a lot.

I am just a free user, and I am an university student. But if I graduated and I had a job, I would definitely donate you guys on Patreon as much as possible. ❤ Really love this software and love you guys' efforts.

You do not have to fix it or debug it right away, I can wait. No need to rush anything.

Again, love you guys for creating this software and hoping your improvement on this software.

Sincerely,

**Comment by Ryosama2** [comment]
<Update Glitch effect: bad news, October 13th, 2022>

Dear developers,

The glitch effects still existed, and I think the AI machine might think that the **glitch effects** were not the bug. But in fact, the **glitch effects** were a real serious bug that caused a negative impact on users.

Let me give you **the log** here: https://thptthaininhh-my.sharepoint.com/:t:/g/personal/minhhlt2000_1driver_vip/EVCSG1v743BClQVk9lr0W5MB2Mg_jw1BJKW_UmbZ4XRr9Q?e=UNlIt9

Next, this is my **original video**, link: https://thptthaininhh-my.sharepoint.com/:v:/g/personal/minhhlt2000_1driver_vip/EWIT4fQl1XROpRPCEHMX17wB-j_589KCLLifBRvPERQNJQ?e=MCGzQe

And this is the video **was interpolated by Flowframes** (contained glitching effects), link: https://thptthaininhh-my.sharepoint.com/:v:/g/personal/minhhlt2000_1driver_vip/Ed-3MeJuEU1Oi4eVwdt3UeUB4POb8iCQ6Xc08NBjxLSByQ?e=UB3d19

And finally, these are **images** that I had taken screenshot in video:
![2022-10-13_220150](https://user-images.githubusercontent.com/115549537/195637232-c4230223-6e76-476d-837a-438417a9b76e.jpg)
![2022-10-13_220232](https://user-images.githubusercontent.com/115549537/195637243-8b8721f3-b448-48d6-bc2a-4d14d143847d.jpg)
![2022-10-13_220342](https://user-images.githubusercontent.com/115549537/195637246-c6ebe37a-9ee6-41ce-b1c8-4947ece2ffda.jpg)
![2022-10-13_220437](https://user-images.githubusercontent.com/115549537/195637248-53467866-b0b3-4927-b936-2876db6d2726.jpg)
![2022-10-13_215831](https://user-images.githubusercontent.com/115549537/195637251-7b626692-266b-44a2-8e83-3a06cfd6443f.jpg)
![2022-10-13_215944](https://user-images.githubusercontent.com/115549537/195637253-ffea1dc0-0788-44fc-bc75-d25673acf234.jpg)

**To sum up, you guys created amazing software. I really love it. The interpolation is magnificent and even awesome.** 
Unfortunately, this is the moment I had a little sad experience with Flowframes. Sorry for this bad news guys.

Sincerely,

**Comment by Ryosama2** [comment]
> 

![image](https://user-images.githubusercontent.com/115549537/195741881-08862b07-0370-46be-a09e-24f4bcccd137.png)

I deleted GPU driver and now I am going to the GPU NVIDIA website for re-installing again

I am a head to https://www.nvidia.com/download/index.aspx# 

It said two options for me to download:

- If you are a gamer who prioritizes day of launch support for the latest games, patches, and DLCs, choose **Game Ready Drivers**.
- If you are a content creator who prioritizes stability and quality for creative workflows including video editing, animation, photography, graphic design, and livestreaming, choose **Studio Drivers**.

![image](https://user-images.githubusercontent.com/115549537/195742187-7c3054a9-19c6-4026-9c27-07a50f3f28f0.png)
![image](https://user-images.githubusercontent.com/115549537/195742239-65a70e92-598d-4ca9-a740-c8319eab75c3.png)

Then which one should I pick?

Thanks for your help, again❤️

**Comment by Ryosama2** [comment]
> Try re-installing your GPU drivers and make sure you did not overclock your GPU, CPU, or RAM

Actually, **I did not even know to how to overlock CPU, GPU or RAM**. I have not ever used those functions at all. **But what if my laptop automatically active the "overlock" function without my permission?**

**Comment by Vaenir** [comment]
Pick the studio driver, and if you have the same issues, then we can probably rule that out.

Automatic overclocks or 'boost clock' shouldn't impact anything, as it does this within a window that is deemed to be safe as hardware will usually push a low more power to make it stable. 

If it is a concern, though, you can disable boost clocks on nvidia cards with MSI Afterburner by pressing CTRL+F in the main program window, then selecting the dot that matches voltage and frequency for the base clock and pressing L. This will lock it to the base clock and it will no longer boost up, but will also no longer reduce clocks and voltage to conserve power, so remember to disable it by pressing L again after you've confirmed it's no longer a problem. 

Additionally, there's a good chance it could be artifacts introduced while encoding rather than during interpolation, so try using a different encoder first. Try using h264, or h265 (not the NVENC ones), as these use the CPU to encode and will have less chance of introducing artifacts while encoding.

**Comment by Ryosama2** [comment]
> Pick the studio driver, and if you have the same issues, then we can probably rule that out.
> 
> Automatic overclocks or 'boost clock' shouldn't impact anything, as it does this within a window that is deemed to be safe as hardware will usually push a low more power to make it stable.
> 
> If it is a concern, though, you can disable boost clocks on nvidia cards with MSI Afterburner by pressing CTRL+F in the main program window, then selecting the dot that matches voltage and frequency for the base clock and pressing L. This will lock it to the base clock and it will no longer boost up, but will also no longer reduce clocks and voltage to conserve power, so remember to disable it by pressing L again after you've confirmed it's no longer a problem.
> 
> Additionally, there's a good chance it could be artifacts introduced while encoding rather than during interpolation, so try using a different encoder first. Try using h264, or h265 (not the NVENC ones), as these use the CPU to encode and will have less chance of introducing artifacts while encoding.
────────────────────────────────────────────────
Thank you for answering.

I have installed both drivers (Game Ready Driver and Studio Driver), just in case. 
![2022-11-14_172932](https://user-images.githubusercontent.com/115549537/201646311-236412eb-bb9e-4fe7-8ad4-47cce46189ca.jpg)

Luckily, NVIDIA Experience did have an option for having both drivers, so I selected "**_Studio Driver_**" mode for interpolating video frame rate. **And it solved well! No more glitching effects.** Many thanks, again! ^^ 

In terms of the "overclock" issue. I still don't know to deal with it next time. I am not using the MSI brand, instead, I am using an Acer laptop. So, this is my laptop configuration (Main laptop model: **Laptop Acer Nitro 5 AN515 45 R9SC**)
![image](https://user-images.githubusercontent.com/115549537/201646991-fda26929-a59d-4121-a046-80e68619ac46.png)

In terms of video encoder, I am currently using H265, since the output quality of the video is better than H264 format.
![image](https://user-images.githubusercontent.com/115549537/201647592-9acf7d6d-3693-46b4-aef1-4e0b91ac330a.png)
